### ISSUE TITLE
Injecting custom repository throws error

### ISSUE THREAD

**Issue by thasmo** [analogueorm/analogue#5]
When Laravel tries to auto-inject a dependent custom repository class I get `Unresolvable dependency resolving [Parameter #0 [ <required> $dsn ]] in class PDO`. Seems like that Laravel tries to recursively inject dependencies into the classes constructors and fails for the PDO class.

**Comment by RemiCollin** [comment]
When auro resolving, Laravel needs to know how to create the repository object, that is, in this case, to inject the correct mapper into the repository. 

Example :  

AcmeServiceProvider :
```
public function register()
{
    $this->app->bind('Acme\Users\UserRepository', function ($app) {
        	$mapper = $app->make('analogue')->mapper('Acme\Users\User');
        	return new \Acme\Users\UserRepository($mapper);
   });
}
```

**Comment by thasmo** [comment]
@RemiCollin, I see. But is there a way to let the Laravel Service Container create a new instance of `Acme\Users\UserRepository` without the need of providing an already instantiated Mapper? Or is this technically not possible because?

**Comment by RemiCollin** [comment]
Well as a shortcut I guess you could instantiate the Mapper directly inside your repository constructor method. Didn't try it this way but should work.

**Comment by RemiCollin** [comment]
I've been wrapping my head a lot recently on how to streamline this process to make it the most user friendly possible. 

In L5, the app() helpers comes in handy : 

```php

class UserRepository {

    public function __construct()
    {
        $mapper = app('analogue')->mapper(User::class);
        parent::__construct($mapper);
    }

}

This way you can have a Repository class that autoresolve without the need to go through the Ioc Container registering process.

**Comment by RemiCollin** [comment]
Fixed in version 2.1

Repositories can now be instanciated with an Entity instance :

```php
class UserRepository {

    public function __construct()
    {
        parent::__construct(new User);
    }
}
```
